baseplate/lib/metrics.py

```diff
@@ -49,20 +49,33 @@
 import time
 
 from types import TracebackType
+from typing import Any
 from typing import DefaultDict
 from typing import Dict
 from typing import List
 from typing import Optional
 from typing import Type
 
+
 from baseplate.lib import config
 
 
 logger = logging.getLogger(__name__)
 
 
 def _metric_join(*nodes: bytes) -> bytes:
-    return b".".join(node.strip(b".") for node in nodes)
+    return b".".join(node.strip(b".") for node in filter(None, nodes))
+
+
+def _format_tags(tags: Optional[Dict]) -> str:
```

Comment:
Since the metrics code is called a _lot_ in an application and I've seen it show up in performance profiling in the past, I was curious if this was the most efficient way to serialize the tags. Particularly since the standard wisdom is that f-strings and `str.join` are faster than `+` concatenation. The results were kinda surprising! 

https://gist.github.com/spladug/b936c5a9528ced0d1095a01aa1136075

So if we want this to be as quick as possible, something like the following would appear good:

```python
def format_tags_concatenate_fstrings(tags: Optional[Dict]) -> str:
    if not tags:                                                  
        return ""                                                 
                                                                  
    formatted_str = "|#"                                          
    for key, value in tags.items():                               
        formatted_str += f"{key}:{value},"                        
    return formatted_str[:-1]                                     
```

(it's a tiny bit faster without stripping the trailing comma off, but I think wasting bytes in our network packets is probably worse)

Feel free to play around with benchmarking and find a faster one too :)